lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -2318,7 +2324,9 @@ private function executeInserts($documents)
                     && !$class->isNullable($fieldName)
                     && !$this->isAutocreatedProperty($class, $fieldName)
                 ) {
-                    throw new PHPCRException(sprintf('Field "%s" of class "%s" is not nullable', $fieldName, $class->name));
+                    if (!$this->treeOpQueue->isQueued(TreeOperation::OP_REMOVE, $oid)) {
+                        throw new PHPCRException(sprintf('Field "%s" of class "%s" is not nullable', $fieldName, $class->name));
```

Comment:
We allow null fields on operations that are queued for removal because the changeset has been computed and unsert the fields, however we still need the node to be created in the timeline (it will be later removed).